core/primitives/piece/piece_data.hpp

```diff
@@ -28,8 +29,11 @@ namespace fc::primitives::piece {
 
     bool isOpened() const;
 
+    bool isNullData() const;
+
    private:
     int fd_;
+    bool is_null_data_;
```

Comment:
We can tell if file is opened or not from `fd_`.  
Do we add `is_null_data_` to separate "file not opened" (and "file opened") from "zeroes" state?  
Should default constructor (and moving) make "zeroes" or "file not opened"?

```suggestion
    static PieceData makeNull() {
      PieceData pd;
      pd.is_null_data_ = true;
      return pd;
    }

   private:
    int fd_{kUnopenedFileDescriptor};
    bool is_null_data_{false};
```

Reply:
Yes, default constructor and moving should make "zeroes" and "file not opened"

Reply:
I asked, if "zeroes" {fd=-1, null=true} and "not open" {fd=-1, null=false} should be considered separate states.  

If "not open" (e.g. open() returned -1, or default constructed, or moved out) = "zeros", then `is_null_data_` is not needed because `isNullData()` = `isOpened()`